vocabularies/Org.OData.Aggregation.V1.xml

```diff
@@ -209,72 +209,210 @@
+        <Annotation Term="Core.LongDescription">
+          <String>Every recursive hierarchy function defined in this vocabulary has a parameter `Hierarchy` of this type.
+This is the qualifier of a `RecursiveHierarchy` annotation on the entity type of
+- the function's binding parameter or
+- the entity set specified by the `HierarchyNodes` parameter, if this is present.
+
+The `RecursiveHierarchy/ParentNavigationProperty` from this annotation defines a hierarchy on the
+binding parameter's entity set or on `HierarchyNodes` if present. This hierarchy is evaluated by the function.
+Nodes in the hierarchy are identified by the primitive value of their `RecursiveHierarchy/NodePropertyPath`.
+If the binding parameter is null, the function returns null.</String>
+        </Annotation>
+      </TypeDefinition>
+
+      <TypeDefinition Name="HierarchyNode" UnderlyingType="Edm.String">
+        <Annotation Term="Core.Description" String="Property path identifying the node in a recursive hierarchy" />
+        <Annotation Term="Core.LongDescription">
+          <String>Every recursive hierarchy function defined in this vocabulary has
+- an overload with a parameter `NodePropertyPath` of this type plus a `HierarchyNodes` parameter and
+- an overload without these two parameters.
+
+In the overload with these parameters, both the entity given in the function's binding parameter
+and its `HierarchyNodes` parameter must have structural properties with the path given in the `NodePropertyPath` parameter
+and with the same type. This is the property containing the entity's node value in the hierarchy.
+The entity need not belong to `HierarchyNodes`, only its node value is compared to the node values in `HierarchyNodes`.</String>
+        </Annotation>
+      </TypeDefinition>
+
       <Function Name="isroot" IsBound="true">
-        <Annotation Term="Core.Description" String="Returns true if and only if the value of the node property of the specified hierarchy is the root of the hierarchy" />
-        <Parameter Name="Entity" Type="Edm.EntityType" Nullable="false" />
-        <Parameter Name="Hierarchy" Type="Edm.String" Nullable="false" />
-        <ReturnType Type="Edm.Boolean" Nullable="false" />
+        <Annotation Term="Core.Description" String="Is the entity the root of the specified hierarchy?" />
+        <Parameter Name="Entity" Type="Edm.EntityType" Nullable="true" />
+        <Parameter Name="Hierarchy" Type="Aggregation.HierarchyQualifier" Nullable="false" />
+        <ReturnType Type="Edm.Boolean" Nullable="true" />
+      </Function>
+      <Function Name="isroot" IsBound="true">
+        <Annotation Term="Core.Description" String="Is the entity the root of the specified hierarchy?" />
+        <Parameter Name="Entity" Type="Edm.EntityType" Nullable="true" />
+        <Parameter Name="Hierarchy" Type="Aggregation.HierarchyQualifier" Nullable="false" />

```

Comment:
```suggestion
        <Parameter Name="Hierarchy" Type="Edm.String" Nullable="false">
          <Annotation Term="Core.Description" String="Navigation property path in `HierarchyNodes` that navigates to a node's parent and thus defines the recursive hierarchy"/>
        </Parameter>
```
I find it misleading to have a qualifier that identifies a `RecursiveHierarchy` annotation, because in this overload, only the `RecursiveHierarchy/ParentNavigationProperty` matters whereas the `RecursiveHiearchy/NodeProperty` is disregarded in favor of the `NodePropertyPath` parameter. Why not simply identify the hierarchy through its parent navigation property?

And if the `RecursiveHierarchy` annotation is only needed in half the overloads, can't we abolish it altogether?